src/graphic/helper/roundSector.ts

```diff
@@ -89,6 +82,46 @@ function computeCornerTangents(
     };
 }
 
+function calcCircleCenter(x: number, y: number, r: number, angle: number) {
+    return {
+        x: x + r * mathCos(angle),
+        y: y + r * mathSin(angle)
+    };
+}
+
+// For compactibility, don't use normalizeCssArray
+// 5 represents [5, 5, 5, 5]
+// [5] represents [5, 5, 0, 0]
+// [5, 10] represents [5, 5, 10, 10]
+// [5, 10, 15] represents [5, 10, 15, 15]
+// [5, 10, 15, 20] represents [5, 10, 15, 20]
+function normalizeCornerRadius(
+    cr: number | string | (number | string)[],
+    r0: number,
+    r: number
+) {
+    let arr: (number | string)[];
+    if (isArray(cr)) {
+        const len = cr.length;
+        if (len === 4) {
+            arr = cr;
+        }
+        else if (len === 3) {
+            arr = cr.concat(cr[len - 1]);
+        }
+        else if (len === 2) {
+            arr = [cr[0], cr[0], cr[1], cr[1]];
+        }
+        else {
+            arr = [cr[0], cr[0], 0, 0];
+        }
+    }
+    else {
+        arr = [cr, cr, cr, cr];
+    }
+    return map(arr, (cr, idx) => parsePercent(cr, idx < 2 ? r0 : r));
```

Comment:
Should we support percentage value on the sector `cornerRadius`? Seems `Rect` didn't support it.

Fixed `cornerRadius` seems more common in case. I will prefer keep it simpler here

Reply:
The percentage value is supported from the very beginning of this feature([see here](https://echarts.apache.org/option.html#series-pie.itemStyle.borderRadius)), but previously this logic was put into the upper ECharts, so I just keep it here for the backward compatibility.

As for the necessity of supporting percentage value, I would think out the case in which the developer hopes to get a rounded sector(like a capsule or sausage?) with the percentage of '50%'. Just like the rectangle, we used to set the `border-radius` to be `50%` to get a circle. If the percentage value is not supported, the developer will have to calculate the half value via their specified `r/r0`. That doesn't look so convenient.

Reply:
UPDATE: It usually won't reach half of the radius, as we restrict the max value of the corner radius...

Reply:
Didn't notice it has been supported. But it's a bit confusing if it use percentage from `r0` and `r1` respectively. From my side it should be `r1 - r0` if we want `'50%'` can be a perfect value to get a round cap on both side. Is it by design that the corner radius of inner side will be different from the outer side if we use a percentage value?

Reply:
Yes. If it's annular, `r1 - r0` is the real relative value, otherwise, it's `r` or `r0`. Perhaps, we should use `r1 - r0` to avoid confusion when using a percentage value.

Reply:
Agree.

Reply:
Fixed.